go-controller/pkg/testing/libovsdb/matchers.go

```diff
@@ -47,10 +48,16 @@ func testDataDeepValueEqualIgnoringUUID(v1, v2 reflect.Value, ignoreUUID bool) b
 		}
 		return reflect.DeepEqual(v1.Elem().Interface(), v2.Elem().Interface())
 	case reflect.Struct:
+		objectType := v1.Type()
 		for i, n := 0, v1.NumField(); i < n; i++ {
-			if tag := v1.Type().Field(i).Tag.Get("ovsdb"); tag == "_uuid" {
+			if tag := objectType.Field(i).Tag.Get("ovsdb"); tag == "_uuid" {
 				continue
 			}
+			if objectType == reflect.TypeOf(nbdb.LogicalRouter{}) {
+				if tag := objectType.Field(i).Tag.Get("ovsdb"); tag == "ports" || tag == "static_routes" {
+					return reflect.ValueOf(v1.Field(i).Interface()).Len() == reflect.ValueOf(v2.Field(i).Interface()).Len()
```

Comment:
The issue is that `LogicalRouter.ports` or `LogicalRouter.static_routes` are uuid references to either `LogicalPort` or `LogicalRouterStaticRoute`. I can't validate that in the tests, since their are runtime generated. So the idea I have is skip validating the actual field, but validate the length expected. Thus you can write a test as: "I expect the `LogicalRouter.static_routes` to have 3 items, but I don't care what they are"

Reply:
I see, I will try to think of something generic.